runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/Engine.java

```diff
@@ -197,24 +221,25 @@ public long counter(
         return counter.applyAsLong(name);
     }
 
-    public Future<Void> start()
+    public Future<NamespaceConfig> start() throws ExecutionException, InterruptedException
     {
         for (AgentRunner runner : runners)
         {
             AgentRunner.startOnThread(runner, Thread::new);
         }
-
-        this.configureRef = commonPool().submit(configure);
-
-        return configureRef;
+        Future<NamespaceConfig> register = commonPool().submit(registerTask);
+        NamespaceConfig rootNamespace = register.get();
+        watcherTask.setRootNamespace(rootNamespace);
+        commonPool().submit(watcherTask);
```

Comment:
I think we need to simplify the flow here a bit more.
The `watcherTask` should be in charge of noticing a new configuration, even at startup.
When it detects a change, the reaction should be established here, with an unregister of the old config (possible no-op) and a register of the new config (possible no-op).
Use a listener callback approach on `watcherTask` so we can easily setup the same reactions for http vs file or any other watcher implementation.

Reply:
I actually started a refactor similar to this, when did the rebase on the HTTP watcher. You're right, the code is mutch cleaner this way.